### PR TITLE
convox run handles non-tty stdin

### LINKED ISSUE #108: panic after convox run
In my CD pipeline, I'm running the `convox` CLI from within nodejs.  It works (the run completes, I get valid output from the docker run), but it dies with the stack below.

test script:
```
var exec = require('child_process').exec;
exec("convox run web 'ls -al'", function (err, stdout, stderr) {
  console.log('STDOUT', stdout);
  console.log('STDERR', stderr);
  if (err) {
    console.error(err);
    process.exit(1);
  }
});
```

sample output:
```
13:33 $ node test.js 
STDOUT total 152
drwxr-xr-x 18 root root 4096 Oct 15 19:57 .
drwxr-xr-x 25 root root 4096 Oct 15 20:35 ..
-rw-------  1 root root  110 Oct 15 19:26 .buildpacks
[... snip ...]

STDERR panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x10d06f]

goroutine 1 [running]:
golang.org/x/crypto/ssh/terminal.Restore(0x0, 0x0, 0x0, 0x0)
	/go/src/golang.org/x/crypto/ssh/terminal/util.go:70 +0xcf
golang.org/x/crypto/ssh/terminal.Restore(0x0, 0x0, 0x0, 0x0)
	/go/src/golang.org/x/crypto/ssh/terminal/util.go:70 +0xcf
main.cmdRun(0x8208b7e60)
	/go/src/github.com/convox/rack/cmd/convox/run.go:60 +0x50c
github.com/codegangsta/cli.Command.Run(0x55ae20, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5a9b20, 0x13, 0x5efd00, ...)
	/go/src/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/codegangsta/cli.(*App).Run(0x8208b7c20, 0x8208b2380, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/codegangsta/cli/app.go:173 +0xfb4
main.main()
	/go/src/github.com/convox/rack/cmd/convox/main.go:40 +0x84

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 17 [syscall]:
syscall.Syscall(0x3, 0x0, 0x820a42000, 0x8000, 0x1c800000002, 0x13, 0x1)
	/usr/local/go/src/syscall/asm_darwin_amd64.s:16 +0x5
syscall.read(0x0, 0x820a42000, 0x8000, 0x8000, 0x8820a990d0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_darwin_amd64.go:972 +0x5f
syscall.Read(0x0, 0x820a42000, 0x8000, 0x8000, 0x820a42000, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0x8208d8008, 0x820a42000, 0x8000, 0x8000, 0x499d20, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x75
os.(*File).Read(0x8208d8008, 0x820a42000, 0x8000, 0x8000, 0x8000, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
io.copyBuffer(0x8820ad9768, 0x820a1c2d0, 0x8820a5d9e0, 0x8208d8008, 0x820a42000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:381 +0x247
io.Copy(0x8820ad9768, 0x820a1c2d0, 0x8820a5d9e0, 0x8208d8008, 0x3a, 0x0, 0x0)
	/usr/local/go/src/io/io.go:351 +0x64
created by github.com/convox/rack/client.(*Client).Stream
	/go/src/github.com/convox/rack/client/client.go:344 +0x901

{ [Error: Command failed: panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x10d06f]

goroutine 1 [running]:
golang.org/x/crypto/ssh/terminal.Restore(0x0, 0x0, 0x0, 0x0)
	/go/src/golang.org/x/crypto/ssh/terminal/util.go:70 +0xcf
golang.org/x/crypto/ssh/terminal.Restore(0x0, 0x0, 0x0, 0x0)
	/go/src/golang.org/x/crypto/ssh/terminal/util.go:70 +0xcf
main.cmdRun(0x8208b7e60)
	/go/src/github.com/convox/rack/cmd/convox/run.go:60 +0x50c
github.com/codegangsta/cli.Command.Run(0x55ae20, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5a9b20, 0x13, 0x5efd00, ...)
	/go/src/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/codegangsta/cli.(*App).Run(0x8208b7c20, 0x8208b2380, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/codegangsta/cli/app.go:173 +0xfb4
main.main()
	/go/src/github.com/convox/rack/cmd/convox/main.go:40 +0x84

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 17 [syscall]:
syscall.Syscall(0x3, 0x0, 0x820a42000, 0x8000, 0x1c800000002, 0x13, 0x1)
	/usr/local/go/src/syscall/asm_darwin_amd64.s:16 +0x5
syscall.read(0x0, 0x820a42000, 0x8000, 0x8000, 0x8820a990d0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_darwin_amd64.go:972 +0x5f
syscall.Read(0x0, 0x820a42000, 0x8000, 0x8000, 0x820a42000, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0x8208d8008, 0x820a42000, 0x8000, 0x8000, 0x499d20, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x75
os.(*File).Read(0x8208d8008, 0x820a42000, 0x8000, 0x8000, 0x8000, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
io.copyBuffer(0x8820ad9768, 0x820a1c2d0, 0x8820a5d9e0, 0x8208d8008, 0x820a42000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:381 +0x247
io.Copy(0x8820ad9768, 0x820a1c2d0, 0x8820a5d9e0, 0x8208d8008, 0x3a, 0x0, 0x0)
	/usr/local/go/src/io/io.go:351 +0x64
created by github.com/convox/rack/client.(*Client).Stream
	/go/src/github.com/convox/rack/client/client.go:344 +0x901
] killed: false, code: 2, signal: null }
```

### PR DESCRIPTION
**Pull request by bobzoller**
fixes #108

### CODE REVIEW

File: cmd/convox/run.go [hunk 1]
```diff
@@ -40,7 +40,6 @@ func cmdRun(c *cli.Context) {
 
 	fd := os.Stdin.Fd()
 	stdinState, err := terminal.GetState(int(fd))
-	defer terminal.Restore(int(fd), stdinState)
```
Comment (csquared) [inline cmd/convox/run.go]: IIRC, this was for handling CTRL+C

Comment (csquared) [inline cmd/convox/run.go]: can we conditionally defer??

Comment (bobzoller) [inline cmd/convox/run.go]: ah, got it.  (couldn't figure out what that was for!)  I'll add it back in and put the conditional around it.  and a comment ;)

**Comment by bobzoller** [comment]
@csquared what do you think about this?  extracting a function keeps the logic simple, and you don't need the double `terminal.Restore` because the one defer covers both cases.

**Comment by csquared** [comment]
This looks good- nice use of defer! Will merge when back at keyboard

**Comment by csquared** [comment]
:+1:

**Pull request merged**